langchain/src/document_loaders/s3.ts

```diff
@@ -0,0 +1,95 @@
+import fsDefault from "fs";
+import path from "path";
+import os from "os";
+import { BaseDocumentLoader } from "./base.js";
+import { UnstructuredLoader as UnstructuredLoaderDefault } from "./unstructured.js";
+
+export class S3Loader extends BaseDocumentLoader {
+  private bucket: string;
+
+  private key: string;
+
+  private unstructuredAPIURL: string;
+
+  private _fs: typeof fsDefault;
+
+  private _UnstructuredLoader: typeof UnstructuredLoaderDefault;
+
+  constructor(
+    bucket: string,
+    key: string,
+    unstructuredAPIURL: string,
+    _fs = fsDefault,
+    _UnstructuredLoader = UnstructuredLoaderDefault
+  ) {
+    super();
+    this.bucket = bucket;
+    this.key = key;
+    this.unstructuredAPIURL = unstructuredAPIURL;
+    this._fs = _fs;
+    this._UnstructuredLoader = _UnstructuredLoader;
+  }
+
+  public async load() {
+    const { S3Client, GetObjectCommand } = await S3LoaderImports();
+
+    const tempDir = this._fs.mkdtempSync(
```

Comment:
I think we'd want to not use `fs` and work with a Buffer in memory. This would enable using this outside of Node, eg in Edge environments

I'm going to open a PR very soon to make the Unstructured loader work with Buffers directly